packages/metro/src/reactNativeTransformer.js

```diff
@@ -47,61 +47,72 @@ const getBabelRC = (function() {
 
     babelRC = {plugins: []};
 
-    // Let's look for the .babelrc in the project root.
+    // Let's look for a babel config file in the project root.
     // In the future let's look into adding a command line option to specify
     // this location.
     let projectBabelRCPath;
+
+    // .babelrc
     if (projectRoot) {
       projectBabelRCPath = path.resolve(projectRoot, '.babelrc');
     }
 
-    // If a .babelrc file doesn't exist in the project,
+    // .babelrc.js
+    if (projectBabelRCPath && !fs.existsSync(projectBabelRCPath)) {
+      projectBabelRCPath = path.resolve(projectRoot, '.babelrc.js');
+    }
+
+    // babel.config.js
+    if (projectBabelRCPath && !fs.existsSync(projectBabelRCPath)) {
+      projectBabelRCPath = path.resolve(projectRoot, 'babel.config.js');
+    }
+
+    // We found a babel config file, extend our config off of it
+    if (projectBabelRCPath && fs.existsSync(projectBabelRCPath)) {
+      babelRC.extends = projectBabelRCPath;
+      return babelRC;
+    }
+
+    // Babel config file doesn't exist in the project,
```

Comment:
I wonder if we can get rid of all the following logic and just do `babelRC.extends = require.resolve('metro/rn-babelrc.json');`... would that work? (we may need to tweak the preset path in `metro/rn-babelrc.json`).

Reply:
I thought that as well but I wasn't sure if I should change it too much just in case - as you've mentioned it though I'll give it a go in the morning and push up👍